lib/turn.rb

```diff
@@ -0,0 +1,163 @@
+require './lib/card'
+require './lib/deck'
+require './lib/player'
+require 'pry'
+
+class Turn
+
+    attr_reader :player1,
+                :player2,
+                :spoils_of_war
+    
+    def initialize(player1, player2)
+
+        @player1 = player1
+        @player2 = player2
+        @spoils_of_war = []
+        
+    end
+
+    def type
+        
+        if @player1.deck.rank_of_card_at(0) != @player2.deck.rank_of_card_at(0)
+            :basic
+        elsif @player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0) 
+
+            if not_enough_cards? == FALSE
+
+                if @player1.deck.rank_of_card_at(2) != @player2.deck.rank_of_card_at(2)
+                :war
+                else
+                :mutually_assured_destruction
+                end
+
+            else
+
+                :war
+                
+            end
+            
+
+        end
+    end
+
+    def winner
+
+
+        if type == :basic
+
+                
+            if @player1.deck.rank_of_card_at(0) > @player2.deck.rank_of_card_at(0)
+            @player1
+            else
+            @player2
+            end
+
+        elsif type == :war
+            
+            if not_enough_cards? == FALSE
+
+                if @player1.deck.rank_of_card_at(2) > @player2.deck.rank_of_card_at(2)
+                @player1
+                else
+                @player2
+                end
+
+            else
+
+            not_enough_cards
+
+            end
+
+        end
+
+    end
+
+
+    def not_enough_cards?
+
+        if @player1.deck.cards.length < 3 
+            TRUE
+        elsif @player1.deck.cards.length < 3 
+            TRUE
+        elsif @player2.deck.cards.length < 3 
+            TRUE
+        elsif @player2.deck.cards.length < 3 
+            TRUE
+        else
+            FALSE
+        end
```

Comment:
Lines 79 and 81 as well as 83 and 85 are duplicates. We only need to check a condition once. This could be shortened to:
```ruby
  def not_enough_cards?
    if @player1.deck.cards.length < 3
      TRUE
    elsif @player2.deck.cards.length < 3
      TRUE
    else
      FALSE
    end
```

or this works as well:
```ruby
    if @player1.deck.cards.length < 3 || @player2.deck.cards.length < 3
      TRUE
    else
      FALSE
    end
```

Reply:
I think I accidentally copy pasted that if condition and didn't realize it! Would you say it's better practice to keep conditions on a single line if they can be merged because of the restricting methods to 7 lines requirement that we'll start having later on?

Reply:
I would say it's better practice to combine them, not because of the 7 line limit but because the fewer paths and outcomes we have, the fewer chances for bugs. You could also move that condition to a helper method which would make the whole thing a little easier to read. Something like:
```ruby
def player_has_fewer_than_3_cards?
  @player1.deck.cards.length < 3 || @player2.deck.cards.length < 3
end
```

And then in the `if` statement you could write:
```ruby
if player_has_fewer_than_3_cards?
  true
else
  false
end
```